app/uk/gov/hmrc/gform/models/FormModel.scala

```diff
@@ -151,6 +152,10 @@ case class FormModel[A <: PageMode](
     fc.id.baseComponentId
   }.toSet
 
+  val lookupRegister: Map[BaseComponentId, Register] = allFormComponents.collect { case fc @ IsTextLookup() =>
+    fc.id.baseComponentId -> fc.`type`.cast[Text].map(_.constraint.cast[Lookup].map(_.register)).get.get
```

Comment:
This can be simplified to
```suggestion
    fc.id.baseComponentId -> lookup.register
```
If `IsTextLookup` would return `Option[Lookup]` instead of `Boolean`